cime_config/SystemTests/fsurdatmodifyctsm.py

```diff
@@ -65,54 +65,49 @@ def _create_config_file(self):
+            print("running the fsurdat_modifier tool: ")
+            print("(1) If your ctsm_pylib environment is out of date or you ")
+            print("have not created the ctsm_pylib environment, yet, you may ")
+            print("get past this error by running ./py_env_create ")
+            print("in your ctsm directory and trying this test again. ")
+            print("(2) If conda is not available, install and load conda, ")
+            print("run ./py_env_create, and then try this test again. ")
+            print("(3) If (1) and (2) are not the issue, then you may be ")
+            print("getting an error within the fsurdat_modifier tool itself. ")
+            print("Default error message: ")
+            print(error.output)
+        except:
+            print("ERROR trying to run fsurdat_modifier tool.")
+            raise
 
     def _modify_user_nl(self):
         append_to_user_nl_files(caseroot = self._get_caseroot(),
                                 component = "clm",
                                 contents = "fsurdat = '{}'".format(self._fsurdat_out))
 
-def _get_python_path():
-    """Get path to ncar_pylib's python on cheyenne
-
-    This is needed because we need a python environment that includes xarray
-    and its dependencies. This is currently hard-coded for cheyenne until we
-    come up with a robust way in CIME of ensuring that the correc python
-    environment is loaded.
-
-    """
-    out = subprocess.check_output(['/glade/u/apps/opt/ncar_pylib/ncar_pylib',
-                                   '-l'], universal_newlines=True)
-
-    # First look for a loaded ('L') python
-    path = _find_path_from_pylib_output(out, 'L')
-    # If no loaded python found, look for a default ('D') python
-    if path is None:
-        path = _find_path_from_pylib_output(out, 'D')
-
-    if path is None:
-        raise RuntimeError('No python found')
-
-    return os.path.join(path, 'bin', 'python')
-
-def _find_path_from_pylib_output(ncar_pylib_output, char):
-    """Given line-by-line output from ncar_pylib, return the path to python if found
-
-    Args:
-    - ncar_pylib_output: line-by-line output from ncar_pylib
-    - char: the character to look for in the leading parenthetical expression (typically 'L' or 'D')
-
-    Returns a path to python, or None if not found
-    """
-    # The line of interest looks like the following (for char = 'L'):
-    # (L) ... /path/to/python
-    regex = r'\(' + char + r'\).* (/\S+)'
-    for line in ncar_pylib_output.splitlines():
-        match_line = re.match(regex, line)
-        if match_line:
-            return match_line.group(1)
-
-    return None
+    def _get_conda_env(self):
+        #
+        # Add specific commands needed on different machines to get conda available
+        # Use semicolon here since it's OK to fail
+        #
+        # Remove python and add conda to environment for cheyennne
+        conda_env = "module unload python; module load conda"
```

Comment:
@slevisconsulting what happens when you try running on izumi? That's one advantage of using a conda environment we should be able to get it work on izumi as well.

Reply:
I checked out the branch on izumi and submitted the same test by changing cheyenne to izumi in the suffix:
`./create_test FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.izumi_intel`

First attempt gave this error:
```
module-info mode load in progress
        ModuleCmd_Load.c(213):ERROR:105: Unable to locate a modulefile for 'conda'

        CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
        If your shell is Bash or a Bourne variant, enable conda for the current user with

            $ echo ". /usr/local/anaconda-3.7.0/etc/profile.d/conda.sh" >> ~/.bashrc

        or, for all users, enable conda with

            $ sudo ln -s /usr/local/anaconda-3.7.0/etc/profile.d/conda.sh /etc/profile.d/conda.sh

        The options above will permanently enable the 'conda' command, but they do NOT
        put conda's base (root) environment on PATH.  To do so, run

            $ conda activate

        in your terminal, or to put the base environment on PATH permanently, run

            $ echo "conda activate" >> ~/.bashrc

        Previous to conda 4.4, the recommended way to activate conda was to modify PATH in
        your ~/.bashrc file.  You should manually remove the line that looks like

            export PATH="/usr/local/anaconda-3.7.0/bin:$PATH"

        ^^^ The above line should NO LONGER be in your ~/.bashrc file! ^^^
```
For the second attempt, I tried my usual approach:
```
module unload lang/python/3.7.0
conda activate ctsm_py
```
and got the same error, which makes sense in retrospect, since `module avail` shows no available conda modules.

@ekluzek I'm not confident about following the instructions in the above error msg, but I'm happy to try if you recommend it.

Reply:
@ekluzek and I confirmed that we both get the same behavior on izumi:
`./py_env_create` works
`conda activate ctsm_pylib` works
BUT we both need to `module unload` python for the fsurdat_modifier tool to work correctly.

Regardless, I will implement code in the test that will check for conda ("which conda") and when this fails (as expected when running the test-suite on cheyenne) we will execute module commands and when it passes we won't.